File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.java
class Solution {
    public long squa(long a, long b){
        if(a * a == b) return (a+1) * (a+1);
        else return -1;
    }
    
    public long solution(long n) {
        long answer = -1;
        long tmp = 0;
        for(int i=0;i<=n;i++){
            tmp = squa(i,n);
            if(tmp != -1){
                return tmp;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

long long squa(long long a, long long b){
    if(a * a == b) return (a+1) * (a+1);
    else return -1;
}

long long solution(long long n) {
    long long answer = -1;
    
    for(int i=0;i<=n;i++){
        answer = squa(i,n);
        if(answer != -1){
            return answer;
        }
    }
    
    return answer;
}